src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.To.cs

```diff
@@ -1,19 +1,33 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Threading;
+
 namespace System.Text.Json.Nodes
 {
     public abstract partial class JsonNode
     {
+        /// <summary>Lazily-initialized options that set the <see cref="JsonSerializerOptions.MaxDepth"/> to match the <see cref="JsonWriterOptions"/> default.</summary>
+        private static JsonSerializerOptions? s_unindentedToStringOptions;
+
+        /// <summary>
+        /// Lazily-initialized options that set the <see cref="JsonSerializerOptions.MaxDepth"/> to match the <see cref="JsonWriterOptions"/> default
+        /// and that sets <see cref="JsonSerializerOptions.WriteIndented"/> to true.
+        /// </summary>
+        private static JsonSerializerOptions? s_indentedToStringOptions;
+
         /// <summary>
         ///   Converts the current instance to string in JSON format.
         /// </summary>
         /// <param name="options">Options to control the serialization behavior.</param>
         /// <returns>JSON representation of current instance.</returns>
         public string ToJsonString(JsonSerializerOptions? options = null)
         {
-            using PooledByteBufferWriter output = WriteToPooledBuffer(options, options?.GetWriterOptions() ?? default);
-            return JsonHelpers.Utf8GetString(output.WrittenMemory.Span);
+            options ??=
+                s_unindentedToStringOptions ??
+                Interlocked.CompareExchange(ref s_unindentedToStringOptions, new JsonSerializerOptions { MaxDepth = JsonWriterOptions.DefaultMaxDepth }, null) ??
+                s_unindentedToStringOptions;
+            return JsonSerializer.Serialize(this, options);
```

Comment:
@eiriktsarpalis, this is running afoul of the trimmer. What's the right / best way to specify the necessary converters and silence the warnings?  Presumably I don't want to use the source generator in System.Text.Json itself to create a context for JsonNode.

Reply:
I don't think using `JsonSerializer` is a viable solution here. It roots all reflection components and changes the behavior of the `ToJsonString()` method so that it defaults to reflection when no options parameter or an unconfigured options parameter is being passed.

> What's the right / best way to specify the necessary converters

The problem is that `JsonNode` (more specifically `JsonValue<T>`) can encapsulate arbitrary .NET values, as such there is no fixed set of contracts we need to supply to make the method happy.

I think you might be able to get similar performance gains if you rewrite the `WriteToPooledBuffer` method to use the https://github.com/dotnet/runtime/blob/e8c3052573d17ac1148ec394986046b29a4d7227/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriterCache.cs class (which is what JsonSerializer is doing).

Reply:
> I think you might be able to get similar performance gains if you rewrite the WriteToPooledBuffer method

That's what I had first, actually, but I simplified it down to just using JsonSerializer since it seemed like unnecessary duplication.  But if using JsonSerializer isn't viable due to trimming-related reasons, that's unfortunate but I can revert it back.

Reply:
I'll close this and open a new PR from my other branch with those changes.